### PR TITLE
Add changelog file for git when doing an update

### PR DESCRIPTION
**Pull request by Nikokrock**
Also remove any output when no change is detected.

### CODE REVIEW

File: e3/anod/checkout.py [hunk 1]
```diff
@@ -7,15 +7,28 @@
...
+
+    When a checkout manager is used in working_dir directory for a repository
+    called name. The following structure will be found on disk after an update
+
+        working_dir/name/               The repository checkout
+                   /name_checkout.json  A json containing some metadata: name,
+                                        url, old_commit, new_commit, revision
+                   /name_changelog.json A json containing the list of commit
+                                        between two call to update. If this is
+                                        the inital checkout or there are not
```
Comment (enzbang) [inline e3/anod/checkout.py]: no changes

File: e3/anod/checkout.py [hunk 2]
```diff
@@ -161,6 +181,20 @@ def update_git(self, url, revision):
             if old_commit == new_commit:
                 result = ReturnValue.unchanged
             else:
+                # Fetch the change log and dump it into the changelog file
+                with open(self.changelog_file, 'w') as fd:
+                    g.write_log(
+                        fd, rev_range='%s..%s' % (old_commit, new_commit))
+
+                commits = []
```
Comment (enzbang) [inline e3/anod/checkout.py]: remove that line, not needed

File: e3/anod/checkout.py [hunk 3]
```diff
@@ -161,6 +181,20 @@ def update_git(self, url, revision):
             if old_commit == new_commit:
                 result = ReturnValue.unchanged
             else:
+                # Fetch the change log and dump it into the changelog file
+                with open(self.changelog_file, 'w') as fd:
```
Comment (enzbang) [inline e3/anod/checkout.py]: That's strange to write some contents in a file and then rewrite it. I would use a temp file for that first dump to make the code easier to read.

File: e3/vcs/git.py [hunk 4]
```diff
@@ -126,7 +126,12 @@ def init(self, url=None, remote='origin'):
         :raise: GitError
         """
         e3.fs.mkdir(self.working_tree)
-        self.git_cmd(['init'])
+        try:
+            result = self.git_cmd(['init'], output=PIPE)
+        except GitError:
+            logger.error('git init output:\n%s', result.out)
+            raise
```
Comment (enzbang) [inline e3/vcs/git.py]: you should not need that given that exceptions and logs are handled by git_cmd

File: e3/anod/checkout.py (hunk 1 above)
Comment (Nikokrock) [reply e3/anod/checkout.py]: taken into account

File: e3/anod/checkout.py (hunk 2 above)
Comment (Nikokrock) [reply e3/anod/checkout.py]: done

File: e3/vcs/git.py (hunk 4 above)
Comment (Nikokrock) [reply e3/vcs/git.py]: done

File: e3/anod/checkout.py [hunk 5]
```diff
@@ -161,6 +183,22 @@ def update_git(self, url, revision):
             if old_commit == new_commit:
                 result = ReturnValue.unchanged
             else:
+                # Fetch the change log and dump it into the changelog file
+                with closing(tempfile.NamedTemporaryFile(mode='w',
+                                                         delete=False)) as fd:
+                    g.write_log(
```
Comment (enzbang) [inline e3/anod/checkout.py]: add a try: finally: to remove the file

**Pull request merged**